ToolBox/KaikoDataConverter/KaikoDataConverterProgram.cs

```diff
@@ -13,42 +13,103 @@
-                CreateCryptoTicks(sourceDirectory, market, tickType, ParseKaikoTradeFile);
-                AggregateTicksInAllResolutions(market, tickType);
+                var processingDate = DateTime.ParseExact(date, DateFormat.EightCharacter, CultureInfo.InvariantCulture);
+                Log.Trace($"KaikoDataConverter(): Starting data conversion from source {filePath.Name} for date {processingDate:yyyy_MM_dd}... ");
+                var zip = new ZipFile(filePath.FullName);
+                var targetDayEntries = zip.Entries.Where(e => e.FileName.Contains($"{processingDate:yyyy_MM_dd}"));
+
+                if (targetDayEntries.Count() == 0)
+                {
+                    Log.Error($"KaikoDataConverter(): Date {processingDate:yyyy_MM_dd} not found in source file.");
+                }
+
+                foreach (var zipEntry in targetDayEntries)
+                {
+                    var nameParts = zipEntry.FileName.Split(new char[] { '/' }).Last().Split(new char[] { '_' });
+                    var exchange = nameParts[0] == "Coinbase" ? "GDAX" : nameParts[0];
+                    var ticker = nameParts[1];
+                    var tickType = nameParts[2] == "trades" ? TickType.Trade : TickType.Quote;
+                    var symbol = Symbol.Create(ticker, SecurityType.Crypto, exchange);
+
+                    var rawData = KaikoCryptoreader.GetRawDataFromEntry(zipEntry);
+                    var ticks = tickType == TickType.Trade ? ParseKaikoTradeFile(rawData, symbol) : ParseKaikoQuoteFile(rawData, symbol);
+                    Log.Trace($"KaikoDataConverter(): Processing {symbol.Value} {tickType}");
+
+                    try
+                    {
+                        Log.Trace($"KaikoDataConverter(): Starting consolidation for {symbol.Value} {tickType}");
+                        var consolidators = GetHighResolutionDataAggregatorsForTickType(tickType);
+
+                        foreach (var tickBar in ticks)
+                        {
+                            foreach (var consolidator in consolidators)
+                            {
+                                consolidator.Consolidator.Update(tickBar);
+                            }
+                        }
+
+                        Log.Trace($"KaikoDataConverter(): Consolidation finished for {symbol.Value} {tickType}");
+                        Log.Trace($"KaikoDataConverter(): Save minute and second files for {symbol.Value} {tickType}");
+
+                        foreach (var consolidator in consolidators)
+                        {
+                            WriteTicksForResolution(symbol, consolidator.Resolution, tickType, consolidator.Flush());
+                        }
+
+                        Log.Trace($"KaikoDataConverter(): Save tick files for {symbol.Value} {tickType}");
+
+                        var writer = new LeanDataWriter(Resolution.Tick, symbol, Globals.DataFolder, tickType);
+                        writer.Write(ticks);
```

Comment:
This is also a new feature in this PR right?

Reply:
Kind of, previously the ticks were written in the [CreateCryptoTicks](https://github.com/Jay-Jay-D/LeanSTP/blob/a3521831557fc861397958905da05b3df0483e00/ToolBox/KaikoDataConverter/KaikoDataConverterProgram.cs#L150) method.